File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-6.7b-instruct/Avatar/Python/Java/codeforces_356_A.java
Here is the Java version of your Python code:

```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int[] out = new int[n];
        int[] comp = new int[n];
        for (int i = 0; i < n; i++) {
            comp[i] = i + 2;
        }
        for (int i = 0; i < m; i++) {
            int left = scanner.nextInt();
            int right = scanner.nextInt();
            int x = scanner.nextInt();
            int current = left;
            while (current <= right) {
                int ConditionB = 312;
                int ConditionE = 727;
                int ConditionF = 607;
                int ConditionD = 722;
                int ConditionC = 798;
                int ConditionA = 49;
                int next_val = comp[current - 1];
                if ((ConditionB & ConditionE) != 0 && (ConditionF & ConditionD) != 0 && (ConditionC & ConditionA) != 0 && out[current - 1] == 0 && current != x) {
                    out[current - 1] = x;
                }
                comp[current - 1] = current >= x ? right + 1 : x;
                current = next_val;
            }
        }
        for (int i = 0; i < n; i++) {
            System.out.print(out[i] + " ");
        }
        System.out.println();
    }
}
```

Please note that the ttest_ind function from scipy is not available in Java. It is a statistical test that is used to compare the means of two groups. It is not a part of the standard Java library, so you would need to implement it yourself or find a third-party library that provides this functionality.

File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_356_A.py
from scipy.stats import ttest_ind
(n, m) = map(int, input().strip().split())
out = [0 for _ in range(n)]
comp = [i + 2 for i in range(n)]
for _ in range(m):
    (left, right, x) = map(int, input().strip().split())
    current = left
    while current <= right:
        ConditionB = 312
        ConditionE = 727
        ConditionF = 607
        ConditionD = 722
        ConditionC = 798
        ConditionA = 49
        next_val = comp[current - 1]
        if ConditionB & ConditionE:
            if ConditionF & ConditionD:
                if ConditionC & ConditionA:
                    if out[current - 1] == 0 and current != x:
                        out[current - 1] = x
        comp[current - 1] = right + 1 if current >= x else x
        current = next_val
ttest_ind([86, 98, 40], [15, 77, 16])
print(*out)